### PR TITLE
Add Ansible template runner.

### PR DESCRIPTION
**Pull request by lfu**
Depends on https://github.com/ManageIQ/manageiq/pull/18160.
Part of https://github.com/ManageIQ/manageiq-automation_engine/pull/265.

https://bugzilla.redhat.com/show_bug.cgi?id=1591797

@miq-bot add_label enhancement

### CODE REVIEW

**Comment by tinaafitz** [comment]
@miq-bot add_reviewer @bzwei @syncrou

**Comment by Fryguy** [comment]
@bdunne Please review.

**Comment by tinaafitz** [comment]
@bilwei @mkanoor Please review.

File: app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  def current_job_timeout(_timeout_adjustment = 1)
+    @execution_ttl ||=
+      (options[:execution_ttl].present? ? options[:execution_ttl].try(:to_i) : DEFAULT_EXECUTION_TTL) * 60
+  end
+
+  def start
+    time = Time.zone.now
+    update_attributes(:started_on => time)
+    miq_task.update_attributes(:started_on => time)
+    my_signal(false, :launch_ansible_tower_job)
```
Comment (bzwei) [inline app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb]: From now on we need to make sure both launching and checking status run on the correct zone, even correct appliance through queue.

File: app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  def queue_signal(*args, deliver_on: nil)
+    priority = options[:priority] || MiqQueue::NORMAL_PRIORITY
+
+    MiqQueue.put(
+      :class_name  => self.class.name,
+      :method_name => "signal",
+      :instance_id => id,
+      :priority    => priority,
+      :role        => 'ems_operations',
+      :zone        => zone || job_template.manager.my_zone,
```
Comment (lfu) [inline app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb]: @mkanoor @bzwei Does this setting make sense?

Comment (bzwei) [reply app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb]: LTGM. Do you have any concern?

Comment (lfu) [reply app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb]: I just added this line.

**Comment by tinaafitz** [comment]
@bzwei Please review.

**Comment by lfu** [comment]
@miq-bot add_label hammer/no

**Pull request merged**